### PR TITLE
Fix usage of LinkNeighborLoader

### PR DESCRIPTION
**Pull request by daralthus**
When hitting https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/loader/link_neighbor_loader.py#L268 `edge_index` would replace `edge_label_index`, `len(dataset)` would be reported incorrectly and sampling gets out of bounds. Like with val_loader...

This PR changes to the correct arguments to avoid this issue. 

In addition:
- Adds new dataloader configs with defaults that fit on a 15GB GPU
- Adds a working example colab notebook

I didn't touch shuffling, let's chat about that separately.

### CODE REVIEW

**Review by itchingpixels — APPROVED** [review]
👍
